### ISSUE TITLE
C++11 not detected

### ISSUE THREAD

**Issue by dwmw2** [opendnssec/SoftHSMv2#250]
Building on Fedora 24 (GCC 6.1.1) I get lots of complaints about `auto_ptr`:

```
softhsm2-util.cpp:88:6: warning: ‘template<class> class std::auto_ptr’ is deprecated [-Wdeprecated-declarations]
 std::auto_ptr<BotanCryptoFactory> BotanCryptoFactory::instance(NULL);
      ^~~~~~~~
```

Setting `HAVE_CXX11` makes them go away. Shouldn't that be detected automatically? In my configure output I can't see any hint of the bits in `ax_cxx_compile_stdcxx_11.m4` actually being invoked...

```
[dwoodhou@i7 SoftHSMv2]$ sh autogen.sh ; ./configure --with-crypto-backend=botan --with-botan=/usr --disable-gost 
libtoolize: putting auxiliary files in '.'.
libtoolize: copying file './ltmain.sh'
libtoolize: putting macros in AC_CONFIG_MACRO_DIRS, 'm4'.
libtoolize: copying file 'm4/libtool.m4'
libtoolize: copying file 'm4/ltoptions.m4'
libtoolize: copying file 'm4/ltsugar.m4'
libtoolize: copying file 'm4/ltversion.m4'
libtoolize: copying file 'm4/lt~obsolete.m4'
configure.ac:45: installing './compile'
configure.ac:37: installing './missing'
src/bin/dump/Makefile.am: installing './depcomp'
src/bin/keyconv/Makefile.am:16: warning: source file '../../lib/crypto/OSSLComp.cpp' is in a subdirectory,
src/bin/keyconv/Makefile.am:16: but option 'subdir-objects' is disabled
automake: warning: possible forward-incompatibility.
automake: At least a source file is in a subdirectory, but the 'subdir-objects'
automake: automake option hasn't been enabled.  For now, the corresponding output
automake: object file(s) will be placed in the top-level directory.  However,
automake: this behaviour will change in future Automake versions: they will
automake: unconditionally cause object files to be placed in the same subdirectory
automake: of the corresponding sources.
automake: You are advised to start using 'subdir-objects' option throughout your
automake: project, to avoid future incompatibilities.
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /usr/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
configure: sysconfdir set to /etc
configure: localstate set to /var
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking whether gcc understands -c and -o together... yes
checking for style of include used by make... GNU
checking dependency style of gcc... gcc3
checking for g++... g++
checking whether we are using the GNU C++ compiler... yes
checking whether g++ accepts -g... yes
checking dependency style of g++... gcc3
checking for dlopen... no
checking for dlopen in -ldl... yes
checking how to run the C preprocessor... gcc -E
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking pthread.h usability... yes
checking pthread.h presence... yes
checking for pthread.h... yes
checking for ECC support... yes
checking for GOST support... no
checking for FIPS 140-2 mode... no
checking for crypto backend... Botan
checking for /usr/include/botan-1.10/botan/init.h... yes
checking what are the Botan includes... -I/usr/include/botan-1.10
checking what are the Botan libs... -L/usr/lib -lbotan-1.10
checking for Botan >= v1.10.0 ... yes
checking for Botan ECC support... Found P256
checking for Botan GNU MP support... no
configure: WARNING:
====================================================
Botan has not been built with GNU MP (--with-gnump).
This will give negative impact on the performance.
====================================================
checking for Botan RFC5649 support... Cannot find AES key wrap with pad
checking for non-paged memory for secure storage... enabled
checking sys/mman.h usability... yes
checking sys/mman.h presence... yes
checking for sys/mman.h... yes
checking the maximum size that may be locked into memory... 64
configure: WARNING:
======================================================================
SoftHSM has been configured to store sensitive data in non-page RAM
(i.e. memory that is not swapped out to disk). This is the default and
most secure configuration. Your system, however, is not configured to
support this model in non-privileged accounts (i.e. user accounts).

You can check the setting on your system by running the following
command in a shell:

	ulimit -l

If this does not return "unlimited" and you plan to run SoftHSM from
non-privileged accounts then you should edit the configuration file
/etc/security/limits.conf (on most systems).

You will need to add the following lines to this file:

#<domain>	<type>		<item>		<value>
*		-		memlock		unlimited

Alternatively, you can elect to disable this feature of SoftHSM by
re-running configure with the option "--disable-non-paged-memory".
Please be advised that this may seriously degrade the security of
SoftHSM.
======================================================================
checking if building with softhsm2-migrate... no
checking if building database object store backend... no
checking if SQLite3 required... no
checking for pkg-config... /usr/bin/pkg-config
checking for getpwuid_r... yes
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking how to print strings... printf
checking for a sed that does not truncate output... /usr/bin/sed
checking for fgrep... /usr/bin/grep -F
checking for ld used by gcc... /usr/bin/ld
checking if the linker (/usr/bin/ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm -B
checking the name lister (/usr/bin/nm -B) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1572864
checking how to convert x86_64-unknown-linux-gnu file names to x86_64-unknown-linux-gnu format... func_convert_file_noop
checking how to convert x86_64-unknown-linux-gnu file names to toolchain format... func_convert_file_noop
checking for /usr/bin/ld option to reload object files... -r
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for ar... ar
checking for archiver @FILE support... @
checking for strip... strip
checking for ranlib... ranlib
checking command to parse /usr/bin/nm -B output from gcc object... ok
checking for sysroot... no
checking for a working dd... /usr/bin/dd
checking how to truncate binary pipes... /usr/bin/dd bs=4096 count=1
checking for mt... no
checking if : is a manifest tool... no
checking for dlfcn.h... yes
checking for objdir... .libs
checking if gcc supports -fno-rtti -fno-exceptions... no
checking for gcc option to produce PIC... -fPIC -DPIC
checking if gcc PIC flag -fPIC -DPIC works... yes
checking if gcc static flag -static works... yes
checking if gcc supports -c -o file.o... yes
checking if gcc supports -c -o file.o... (cached) yes
checking whether the gcc linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... yes
checking how to run the C++ preprocessor... g++ -E
checking for ld used by g++... /usr/bin/ld -m elf_x86_64
checking if the linker (/usr/bin/ld -m elf_x86_64) is GNU ld... yes
checking whether the g++ linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking for g++ option to produce PIC... -fPIC -DPIC
checking if g++ PIC flag -fPIC -DPIC works... yes
checking if g++ static flag -static works... yes
checking if g++ supports -c -o file.o... yes
checking if g++ supports -c -o file.o... (cached) yes
checking whether the g++ linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking dynamic linker characteristics... (cached) GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating softhsm2.module
config.status: creating src/Makefile
config.status: creating src/lib/Makefile
config.status: creating src/lib/common/Makefile
config.status: creating src/lib/common/softhsm2.conf
config.status: creating src/lib/common/softhsm2.conf.5
config.status: creating src/lib/crypto/Makefile
config.status: creating src/lib/crypto/test/Makefile
config.status: creating src/lib/data_mgr/Makefile
config.status: creating src/lib/data_mgr/test/Makefile
config.status: creating src/lib/object_store/Makefile
config.status: creating src/lib/object_store/test/Makefile
config.status: creating src/lib/session_mgr/Makefile
config.status: creating src/lib/session_mgr/test/Makefile
config.status: creating src/lib/slot_mgr/Makefile
config.status: creating src/lib/slot_mgr/test/Makefile
config.status: creating src/lib/handle_mgr/Makefile
config.status: creating src/lib/handle_mgr/test/Makefile
config.status: creating src/lib/test/Makefile
config.status: creating src/lib/test/softhsm2.conf
config.status: creating src/lib/test/softhsm2-alt.conf
config.status: creating src/lib/test/tokens/dummy
config.status: creating src/bin/Makefile
config.status: creating src/bin/common/Makefile
config.status: creating src/bin/dump/Makefile
config.status: creating src/bin/keyconv/Makefile
config.status: creating src/bin/migrate/Makefile
config.status: creating src/bin/util/Makefile
config.status: creating config.h
config.status: config.h is unchanged
config.status: executing depfiles commands
config.status: executing libtool commands
[dwoodhou@i7 SoftHSMv2]$ 
```

**Comment by bellgrim** [comment]
AX_CXX_COMPILE_STDCXX_11 is only called if we compile with Botan v11. 

Does it work better if you add this to configure.ac? Do not know if it will work.

```
AX_CXX_COMPILE_STDCXX_11([noext],[optional])
```

**Comment by dwmw2** [comment]
Yes, that makes it much happier; thanks.

**Comment by bellgrim** [comment]
Fixed in #251
